elki/src/main/java/ch/ethz/globis/pht/PhTree.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2011-2015 ETH Zurich. All Rights Reserved.
+ *
+ * This software is the proprietary information of ETH Zurich.
+ * Use is subject to license terms.
+ */
+package ch.ethz.globis.pht;
+
+import java.util.List;
+
+import ch.ethz.globis.pht.util.PhIteratorBase;
+import ch.ethz.globis.pht.util.PhTreeQStats;
+import ch.ethz.globis.pht.v8.PhTree8;
+import de.lmu.ifi.dbs.elki.utilities.documentation.Reference;
+
+/**
+ * k-dimensional index (quad-/oct-/n-tree).
+ * Supports key/value pairs.
+ *
+ *
+ * @author ztilmann (Tilmann Zaeschke)
+ *
+ */
+@Reference(authors = "T. Zaeschke, C. Zimmerli, M.C. Norrie", title = "The PH-Tree -- A Space-Efficient Storage Structure and Multi-Dimensional Index", booktitle = "Proc. Intl. Conf. on Management of Data (SIGMOD'14), 2014", url = "http://dx.doi.org/10.1145/361002.361007")
```

Comment:
It may be okay to keep the @Reference annotation only on the "MinimalisticMemoryPHTree" class in the ELKI package, to avoid dependencies from ch.ethz to de.lmu.ifi.dbs.elki. Up to you to decide. Same in PhTree8.java. If we drop these annotations (it will still show up in the generated documentation; but only with one class name instead of three) there should be no dependency ch.ethz.* -> de.lmu.* anymore, and the MinimalisticMemoryPHTree class would be the sole interface between the two codebases.